tests/queries/0_stateless/02233_set_enable_with_statement_cte_perf.sql

```diff
@@ -0,0 +1,27 @@
+DROP TABLE IF EXISTS ev;
+DROP TABLE IF EXISTS idx;
+
+CREATE TABLE ev (a Int32, b Int32) Engine=MergeTree() ORDER BY a;
+CREATE TABLE idx (a Int32) Engine=MergeTree() ORDER BY a;
+INSERT INTO ev SELECT number, number FROM numbers(100000000);
+INSERT INTO idx SELECT number*5 FROM numbers(1000);
+
+SET enable_global_with_statement = 1;
+
+-- test_enable_global_with_statement_performance_1
+WITH 'test' AS u SELECT count()FROM ev WHERE a IN (SELECT a FROM idx);
+
+-- test_enable_global_with_statement_performance_2
+SELECT count() FROM ev WHERE a IN (SELECT a FROM idx);
+
+SET enable_global_with_statement = 0;
+
+-- test_enable_global_with_statement_performance_3
+WITH 'test' AS u SELECT count() FROM ev WHERE a IN (SELECT a FROM idx);
+
+SYSTEM FLUSH LOGS;
+
+SELECT count(read_rows) FROM (SELECT read_rows FROM system.query_log WHERE current_database=currentDatabase() AND type='QueryFinish' AND query LIKE '%test_enable_global_with_statement_performance%' ORDER BY event_date, event_time DESC LIMIT 3) GROUP BY read_rows;
```

Comment:
I'm not sure what is this supposed to be testing, but it's not testing the fix. It will return 3 no matter the CH version, right?

Reply:
Yes, thanks, there is a mistake here.

There are 3 SELECT queries:
1. WITH + enable_global_with_statement=1
2. enable_global_with_statement=1
3. WITH + enable_global_with_statement=0

Before the fix: read_rows is 100M for 1 and 9K for 2 and 3.
After the fix: read_rows is 9K for 1, 2 and 3.

The idea with the last query is to measure the if grouping by read_rows the counter for the first item is 3.

Fixing it.

Reply:
COUNT(read_rows) is OK, I have changed SORT condition to use initial_query_start_time_microseconds.